aptos-move/framework/supra-framework/sources/dora_committee.move

```diff
@@ -0,0 +1,1043 @@
+/// Design:
+/// CommitteeInfoStore: store all the committee information, supported operation: add, remove
+/// CommitteeInfo: store all the dora node information, supported operation: add, remove
+/// DoraNodeInfo: store the dora node information, supported operation: update
+/// NodeData: store the dora node information with operator's address
+/// requirements:
+/// 1. two committee has different id
+/// 2. two indentical node should not be in the same committee
+module supra_framework::dora_committee {
+    use std::error;
+    use std::option;
+    use std::signer;
+    use std::vector;
+    use aptos_std::capability;
+    use supra_framework::event::{emit_event, EventHandle};
+    use aptos_std::simple_map::{Self, SimpleMap};
+    use supra_framework::account::{Self, new_event_handle};
+
+    /// The number of committee is not equal to the number of committee member
+    const INVALID_COMMITTEE_NUMBERS: u64 = 4;
+
+    /// The node is not found in the committee
+    const NODE_NOT_FOUND: u64 = 5;
+
+    /// The committee is not found
+    const INVALID_COMMITTEE_ID: u64 = 6;
+
+    /// The committee type is invalid
+    const INVALID_COMMITTEE_TYPE: u64 = 7;
+
+    /// The number of nodes in the committee is invalid
+    const INVALID_NODE_NUMBERS: u64 = 8;
+
+    /// Define the CommitteeType as constants
+    const FAMILY: u8 = 1;
+    const CLAN: u8 = 2;
+    const TRIBE: u8 = 3;
+
+    const SEED_COMMITTEE: vector<u8> = b"supra_framework::dora_committee::CommitteeInfoStore";
+
+    /// Capability that grants an owner the right to perform action.
+    struct OwnerCap has drop {}
+
+    struct SupraCommitteeEventHandler has key {
+        create: EventHandle<CreateCommitteeInfoStoreEvent>,
+        add_committee: EventHandle<AddCommitteeEvent>,
+        remove_committee: EventHandle<RemoveCommitteeEvent>,
+        update_committee: EventHandle<UpdateCommitteeEvent>,
+        add_committee_member: EventHandle<AddCommitteeMemberEvent>,
+        remove_committee_member: EventHandle<RemoveCommitteeMemberEvent>,
+        update_node_info: EventHandle<UpdateNodeInfoEvent>,
+        update_dkg_flag: EventHandle<UpdateDkgFlagEvent>,
+    }
+
+    struct CommitteeInfoStore has key {
+        // map from unique committee id -> CommitteeInfo
+        committee_map: SimpleMap<u64, CommitteeInfo>,
+        // map from node address to committee to which the node belongs
+        node_to_committee_map: SimpleMap<address, u64>,
+    }
```

Comment:
I discussed with @dhaval-supraoracles, and he mentioned in the [PR comments](https://github.com/Entropy-Foundation/supra-oracles/pull/216#discussion_r1595445947) that we need to add an `aggregator_probability` field in the **CommitteeInfoStore**.
During contract initialization, the default value will be **1000000**, and there will be view and edit functions for it."

Reply:
@nizam-supraoracles That should be easy to add. Why the default value for a probability could be 1000000? I think it should below 1?

Reply:
[Here](https://github.com/Entropy-Foundation/supra-oracles/blob/master/nodes/oracle/src/config/resource/committee_config.json#L6891) is the reason why the default value should be **1000000**, but it can be edited.

Reply:
Upated in https://github.com/Entropy-Foundation/aptos-core/pull/7/commits/28d6a8e065b4347228b615264377397cc52c15fd